jetty-core/jetty-deploy/src/main/java/org/eclipse/jetty/deploy/providers/ContextProvider.java

```diff
@@ -460,10 +216,6 @@ else if (context instanceof Supplier<?> supplier)
                 if (contextHandler == null)
                     throw new IllegalStateException("Unknown context type of " + context);
 
-                // Set the classloader if we have a coreContextClassLoader
-                if (coreContextClassLoader != null)
-                    contextHandler.setClassLoader(coreContextClassLoader);
-
                 return contextHandler;
```

Comment:
Isn't there a missing call to `initializeContextHandler()`? Why are we returning here, rather than continuing down?

Reply:
The initializeContextHandler and subsequent initializeContextPath calls are for when there is no XML context deployable.

Example: the deployable is either a directory or a war file.
I'll try to make the comments more clear about this.